src/app/job-registration-form/job-registration-form.component.spec.ts

```diff
@@ -26,16 +29,111 @@ describe('JobRegistrationFormComponent', () => {
                 HttpClientTestingModule,
                 BrowserAnimationsModule
             ],
+            providers: [JobService],
             schemas: [CUSTOM_ELEMENTS_SCHEMA]
-        }).compileComponents();
-    }));
+        });
 
-    beforeEach(() => {
         fixture = TestBed.createComponent(JobRegistrationFormComponent);
         component = fixture.componentInstance;
         fixture.detectChanges();
+        control = new FormControl();
     });
-    it('should create', () => {
+
+    it('JobRegistrationFormComponent should create', () => {
         expect(component).toBeTruthy();
     });
+
+    it('createForm should create jobForm', () => {
+        component.jobForm = null;
+        component.createForm();
+        expect(component.jobForm.getRawValue().toString()).toBe(
+            {
+                date: null,
+                idea: '',
+                organisation: '',
+                region: '',
+                category: '',
+                email: '',
+                contactName: '',
+                website: '',
+                phone: '',
+                description: ''
+            }.toString()
+        );
+    });
+
+    it('noWhiteSpaceValidator function should return that empty string is invalid', () => {
+        control.setValue('');
+        expect(component.noWhiteSpaceValidator(control)).toBeTruthy();
+    });
+
+    it('noWhiteSpaceValidator function should return that spaces only are invalid', () => {
+        control.setValue('                                   ');
+        expect(component.noWhiteSpaceValidator(control)).toBeTruthy();
+    });
+
+    it('noWhiteSpaceValidator function should return that null is invalid', () => {
+        control.setValue(null);
+        expect(component.noWhiteSpaceValidator(control)).toBeTruthy();
+    });
+
+    it('noWhiteSpaceValidator function should return that text is not considered invalid', () => {
+        control.setValue('valid text');
+        expect(component.noWhiteSpaceValidator(control)).toBe(null);
+    });
+
+    it('addJob should call JobService', () => {
```

Comment:
again, before every function with more than 1 test, it's recommended to add describe block and store all it blocks inside